### PR TITLE
Add developer settings

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Intended to solve https://github.com/Azure/azure-sdk-for-net/issues/10998
Passing `UpdateSource` as a parameter to the build or using environment variable `AZURE_DEV_UPDATESOURCES` should run `Export-API.ps1` target as well as `Update-Snippets.ps1`

### CODE REVIEW

File: eng/Directory.Build.Data.targets [hunk 1]
```diff
@@ -26,12 +26,10 @@
...
   <Target Name="ValidateReleaseNotes" BeforeTargets="SetPackageProjectUrlandReleaseNotes" Condition="'$(IsShippingDataPlaneLibrary)' == 'true'" >
     <PropertyGroup>
-      <PowerShellExe Condition=" '$(PowerShellExe)'=='' ">"%ProgramFiles%\PowerShell\6\pwsh.exe"</PowerShellExe>
-      <PowerShellExe Condition="!Exists('$(PowerShellExe)')">pwsh</PowerShellExe>
       <ValidateReleaseNotesScriptPath Condition=" '$(ValidateReleaseNotesScriptPath)'=='' ">$(MSBuildThisFileDirectory)common/scripts/Verify-ChangeLog.ps1</ValidateReleaseNotesScriptPath>
       <ChangeLogPath>$([MSBuild]::NormalizeDirectory($(MSBuildProjectDirectory)/../))CHANGELOG.md</ChangeLogPath>
     </PropertyGroup>
-    <Exec ContinueOnError="true" ConsoleToMSBuild="true" Command="$(PowerShellExe) -NonInteractive -executionpolicy Unrestricted -File $(ValidateReleaseNotesScriptPath) -ChangeLogLocation $(ChangeLogPath) -VersionString $(_VersionInProject)">
+    <Exec ContinueOnError="true" ConsoleToMSBuild="true" Command="pwsh -NonInteractive -executionpolicy Unrestricted -File $(ValidateReleaseNotesScriptPath) -ChangeLogLocation $(ChangeLogPath) -VersionString $(_VersionInProject)">
```
Comment (weshaggard) [inline eng/Directory.Build.Data.targets:32]: Why this change? This requires that pwsh is already on the path.

File: eng/service.proj [hunk 2]
```diff
@@ -35,4 +35,13 @@
              SkipNonexistentTargets="true" />
   </Target>
 
+  <!--Run Aggregate Updates to Source -->
+  <Target Name="AggregateUpdatesToSource" AfterTargets="Build" Condition="'$(UpdateSource)' == 'true' or '$(AZURE_DEV_UPDATESOURCES)' == 'true'" >
```
Comment (weshaggard) [inline eng/service.proj:39]: While it might not matter yet since there is only this one condition I suggest we set the UpdateSource property based on the AZURE_DEV_UPDATESOURCES env variable assuming it isn't already set. 

```
<PropertyGroup>
  <UpdateSource Condition="'$(UpdateSource)' == ''">$(AZURE_DEV_UPDATESOURCES)<UpdateSource>
</PropertyGroup>
```

File: eng/service.proj [hunk 3]
```diff
@@ -35,4 +35,13 @@
...
+    <PropertyGroup>
+      <UpdateSnippetsScriptPath Condition=" '$(UpdateSnippetsScriptPath)'=='' ">$(MSBuildThisFileDirectory)scripts/Update-Snippets.ps1</UpdateSnippetsScriptPath>
+      <ExportAPIScriptPath Condition=" '$(ExportAPIScriptPath)'=='' ">$(MSBuildThisFileDirectory)scripts/Export-API.ps1</ExportAPIScriptPath>
+    </PropertyGroup>
+    <Exec ConsoleToMSBuild="true" Command="pwsh -NonInteractive -executionpolicy Unrestricted -File $(UpdateSnippetsScriptPath) -ServiceDirectory $(ServiceDirectory)"/>
```
Comment (weshaggard) [inline eng/service.proj:44]: @pakrym @tg-msft are these the only scripts we want folks to run?

File: eng/Directory.Build.Data.targets (hunk 1 above)
Comment (weshaggard) [reply eng/Directory.Build.Data.targets:32]: I suggest we still keep a PowersehllExe property to allow folks to override the path if needed instead of hardcoding it everywhere.

File: eng/service.proj (hunk 2 above)
Comment (pakrym) [reply eng/service.proj:39]: Can we have a more expressive name? UpdateSourceOnBuild?

File: eng/service.proj (hunk 3 above)
Comment (pakrym) [reply eng/service.proj:44]: Export-API and snippets are the main two, yes.

File: eng/service.proj (hunk 2 above)
Comment (pakrym) [inline eng/service.proj:39]: This won't work in VS would it?

File: eng/service.proj (hunk 3 above)
Comment (tg-msft) [reply eng/service.proj:44]: We should add comments here and at https://github.com/Azure/azure-sdk-for-net/blob/master/eng/scripts/CodeChecks.ps1 to keep these lists in sync.

Comment (weshaggard) [reply eng/service.proj:44]: Should we instead run CodeChecks.ps1 in a mode that doesn't fail if there are changes?

File: eng/service.proj (hunk 2 above)
Comment (weshaggard) [reply eng/service.proj:39]: You would need to set the env variable before launching VS from the command line. Or have another way of setting this property from msbuild.

Comment (pakrym) [reply eng/service.proj:39]: Nobody opens the `eng/service.proj` in VS

File: eng/service.proj (hunk 3 above)
Comment (pakrym) [reply eng/service.proj:44]: It also runs code generation that's extremely slow and does some other stuff. We might add a flag to it though to only run these two things.

File: eng/service.proj (hunk 2 above)
Comment (weshaggard) [reply eng/service.proj:39]: Ah yes that is fair. this should go into the common targets file.

File: eng/service.proj (hunk 3 above)
Comment (weshaggard) [reply eng/service.proj:44]: OK I'd suggest adding a flag to codechecks.ps1 to run the subset we would run here. That makes things more consistent to what CI would do as well.

File: eng/SnippetGenerator/Program.cs [hunk 4]
```diff
@@ -16,9 +16,18 @@ public class Program
 
         public void OnExecuteAsync()
         {
-            foreach (var sdkDir in Directory.GetDirectories(BasePath))
```
Comment (weshaggard) [inline eng/SnippetGenerator/Program.cs:19]: As we talked we should file an issue to move the SnippetGenerator to the azure-sdk-tools repo. In the world were folks set this new setting we will have race conditions with people trying to build this project and it is generally a bad idea to build tools that are also being consumed by the build as part of the build.

File: eng/Directory.Build.Data.targets [hunk 5]
```diff
@@ -37,6 +35,14 @@
     <Error Condition="'$(SetReleaseNotesErrorCode)' != '0'" Text="No entry for version [$(_VersionInProject)] found in the ChangeLog [$(ChangeLogPath)]. " />
   </Target>
 
+  <!--Run Aggregate Updates to Source -->
+  <Target Name="UpdateSourceOnBuild" AfterTargets="Build" Condition="'$(UpdateSourceOnBuild)' == 'true' and '$(IsShippingClientLibrary)' == 'true'" >
+    <PropertyGroup>
+      <CodeChecksScriptPath Condition=" '$(CodeChecksScriptPath)'=='' ">$(MSBuildThisFileDirectory)scripts\CodeChecks.ps1</CodeChecksScriptPath>
+    </PropertyGroup>
+    <Exec ConsoleToMSBuild="true" Command="$(PowerShellExe) -NonInteractive -executionpolicy Unrestricted -File $(CodeChecksScriptPath) -ProjectDirectory $(MSBuildProjectDirectory)"/>
```
Comment (MiYanni) [inline eng/Directory.Build.Data.targets:43]: Anytime a pwsh command is called, it should also include the `-NoProfile` flag too. Otherwise, you'll get issues with people who set custom PowerShell profiles (I've created profiles for myself in the past, for example).

**Review by MiYanni — APPROVED** [review]

File: eng/SnippetGenerator/Program.cs (hunk 4 above)
Comment (chidozieononiwu) [reply eng/SnippetGenerator/Program.cs:19]: Issue created here https://github.com/Azure/azure-sdk-for-net/issues/13679

**Comment by weshaggard** [comment]
@chidozieononiwu can you also please add information about this in the contributing.md?

File: eng/Directory.Build.Data.props [hunk 6]
```diff
@@ -47,6 +47,8 @@
     <EnableFxCopAnalyzers Condition="'$(IsShippingClientLibrary)' == 'true'">true</EnableFxCopAnalyzers>
     <EnableStyleCopAnalyzers Condition="'$(EnableStyleCopAnalyzers)' == '' and '$(IsClientLibrary)' == 'true'">true</EnableStyleCopAnalyzers>
     <GenerateAPIListing Condition="'$(IsShippingClientLibrary)' == 'true'">true</GenerateAPIListing>
+    <UpdateSourceOnBuild Condition="'$(UpdateSourceOnBuild)' == ''">$(AZURE_DEV_UPDATESOURCESONBUILD)</UpdateSourceOnBuild>
```
Comment (tg-msft) [inline eng/Directory.Build.Data.props:50]: Could we add a quick description of what `AZURE_DEV_UPDATESOURCESONBUILD` does and valid values for it in Contributing.md or somewhere similar?  I don't think anyone's ever going to find it otherwise.

Comment (chidozieononiwu) [reply eng/Directory.Build.Data.props:50]: Added with this PR https://github.com/Azure/azure-sdk-for-net/pull/13721
